coreclient/src/conversations/persistence.rs

```diff
@@ -138,8 +138,14 @@ impl Conversation {
     ) -> Result<(), rusqlite::Error> {
         for (conversation_id, timestamp) in mark_as_read_data.into_iter() {
             transaction.execute(
-                "UPDATE conversations SET last_read = ? WHERE conversation_id = ?",
-                params![timestamp, conversation_id],
+                "UPDATE conversations 
+                 SET last_read = CASE 
+                                    WHEN last_read IS NULL OR last_read < ? 
+                                    THEN ? 
+                                    ELSE last_read 
+                                 END 
+                 WHERE conversation_id = ?",
+                params![timestamp, timestamp, conversation_id],
```

Comment:
You can avoid repeating parameters like that by using the `named_params` macro instead.